verification/verify.go

```diff
@@ -153,3 +160,75 @@ func IsRetryable(err error) bool {
 	_, retryable := err.(Retryable)
 	return retryable
 }
+
+func verifyPixelParams(params *Params) error {
+	// check if the node is on-chain mode
+	if params != nil &&
+		params.Orchestrator != nil &&
+		params.Orchestrator.TicketParams != nil {
+
+		var err error
+		for i := 0; err == nil && i < len(params.Results.Segments); i++ {
+			rendition := []byte{}
+			if i < len(params.Renditions) {
+				rendition = params.Renditions[i]
```

Comment:
Would we be able to avoid allocating an extra var here if `i < len(params.Renditions` was a part of the for loop conditional? i.e. `... i < len(params.Results.Segments) && i < len(params.Renditions)`

Reply:
If we do that, then you could end up with a false positive.  If, somehow, the number of segments is not equal, then it should not pass just because the first x segments match - it should fail.

Reply:
Optimized in 513a646421c01079297b66f40daa5f18040f7b97 to simply exit the function with `ErrPixelMismatch` when `len(params.Results.Segments) != len(params.Renditions)`